@stellar/design-system-website/src/components/Details/index.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Layout,
+  Heading1,
+  Heading2,
+  TextLink,
+  Heading5,
+} from "@stellar/design-system";
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { ElementCode } from "components/ElementCode";
+import { componentDetails } from "constants/componentDetails";
+import {
+  ComponentExample,
+  ComponentProp,
+  ComponentDetailsId,
+} from "types/types.d";
+
+import "./styles.scss";
+
+interface paramProps {
+  id: ComponentDetailsId;
+}
+
+export const Details = () => {
+  const params: paramProps = useParams();
+
+  useEffect(() => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
+  }, []);
+
+  if (!componentDetails[params.id]) {
+    return (
+      <Layout.Inset>
+        <p>
+          Component <code>{params.id}</code> does not exist
+        </p>
+      </Layout.Inset>
+    );
+  }
+
+  const { title, description, examples, props, externalProps } =
+    componentDetails[params.id];
+
+  const renderPropType = (type: string[]) => {
+    const size = type.length;
+
+    return type.map((item, index) => (
+      <span key={`type-${Math.random()}`}>
+        <code>{item}</code>
+        {size !== index + 1 ? " | " : ""}
+      </span>
+    ));
+  };
+
+  return (
+    <Layout.Inset>
+      {/* heading */}
+      <Heading1>{title}</Heading1>
+
+      {/* description */}
+      {description}
+
+      {/* examples */}
+      <Heading2>Examples</Heading2>
+
+      {examples.map((example: ComponentExample) => {
+        const {
+          title: exampleTitle,
+          description: exampleDescription,
+          component,
+        } = example;
+
+        return (
+          <div className="Details__example" key={`example-${Math.random()}`}>
+            <div className="Details__example__container">
+              <div className="Details__example__details">
+                {exampleTitle ? <Heading5>{exampleTitle}</Heading5> : null}
+                {exampleDescription ? <p>{exampleDescription}</p> : null}
+                <div className="Details__example__component">
+                  {component.map((c) => c)}
+                </div>
+              </div>
+              <div className="Details__example__code">
+                <ElementCode element={component} />
+              </div>
+            </div>
+          </div>
+        );
+      })}
+
+      {/* props */}
+      <Heading2>Props</Heading2>
+
+      <table>
+        <thead>
+          <tr>
+            <th>Prop</th>
+            <th>Type</th>
+            <th>Default</th>
+            <th>Optional</th>
+            <th>Description</th>
+          </tr>
+        </thead>
+        <tbody>
+          {props.map((prop: ComponentProp) => (
```

Comment:
NIT: can we rename `prop` to `componentProp`? Think it'd be a little clearer when you call it `componentProp.prop` rather than `prop.prop`

Reply:
👍 Good point.